pkg/tools/etcd_helper.go

```diff
@@ -155,11 +164,14 @@ func (h *EtcdHelper) ExtractToList(key string, listObj runtime.Object) error {
 // a zero object of the requested type, or an error, depending on ignoreNotFound. Treats
 // empty responses and nil response nodes exactly like a not found error.
 func (h *EtcdHelper) ExtractObj(key string, objPtr runtime.Object, ignoreNotFound bool) error {
+	key = h.prefixEtcdKey(key)
```

Comment:
We need a test case that works against ExtractObj to validate the behavior here.